Add Support for OCI Data Science Embedding Models

mrDzurb opened the pull request:
# Description
* Implement support for invoking OCI DSC Model Deployment Embedding models from LlamaIndex. 
https://jira.oci.oraclecorp.com/browse/ODSC-63451
* Updates the LlamaIndex documentation to include a detailed guide on integrating OCI DSC models, covering the necessary configurations, API connections, and common use cases.
https://jira.oci.oraclecorp.com/browse/ODSC-63453

# Testing
## Installation
* Pull the repo
* ```pip install -e llama-index-integrations/embeddings/llama-index-embeddings-oci-data-science```
* It will also require to install ```oracle-ads```, ```llama-index```

## Usage

```bash
import ads
from llama_index.embeddings.oci_data_science import OCIDataScienceEmbeddings

ads.set_auth(auth="security_token", profile="<replace-with-your-profile>")

embedding = OCIDataScienceEmbeddings(
    endpoint="https://<MD_OCID>/predict",
)

e1 = embeddings.get_query_embedding("This is a test document")
print(e1)

e2 = embeddings.get_text_embedding("This is a test document")
print(e2)

e3 = embeddings.get_text_embedding_batch([
        "This is a test document",
        "This is another test document"
    ])
print(e3)
```

## Async

```bash
import ads
from llama_index.embeddings.oci_data_science import OCIDataScienceEmbeddings

ads.set_auth(auth="security_token", profile="<replace-with-your-profile>")

embedding = OCIDataScienceEmbeddings(
    endpoint="https://<MD_OCID>/predict",
)

e1 = embeddings.aget_text_embedding("This is a test document")
print(e1)

e2 = await embeddings.aget_text_embedding_batch([
        "This is a test document",
        "This is another test document"
    ])
print(e2)
```

## Notebook Examples
[Archive.zip](https://github.com/user-attachments/files/18016582/Archive.zip)

mrDzurb committed:
Add Support for OCI Data Science Embedding Models

mrDzurb committed:
Renames OCIDataScienceEmbeddings to OCIDataScienceEmbedding

darenr commented on llama-index-integrations/embeddings/llama-index-embeddings-oci-data-science/README.md:
```diff
@@ -0,0 +1,74 @@
+e1 = embeddings.get_query_embedding("This is a test document")
+print(e1)
+
+e2 = embeddings.get_text_embedding("This is a test document")
```
don't really need the second identical example

mrDzurb replied:
Agree, removed one.

darenr commented on llama-index-integrations/embeddings/llama-index-embeddings-oci-data-science/README.md:
```diff
@@ -0,0 +1,74 @@
+    endpoint="https://<MD_OCID>/predict",
+)
+
+e1 = embeddings.aget_text_embedding("This is a test document")
```
doesn't this need an `await`?

mrDzurb replied:
Good catch, fixed.

mrDzurb committed:
Fixes the docs and examples

VipulMascarenhas commented on docs/docs/examples/embeddings/oci_data_science.ipynb:
nit: 
- colab link point to bedrock notebook, replace with `oci_data_science.ipynb`.
- replace "%pip install llama-index-embeddings-oci-data-science" with "!%pip install llama-index-embeddings-oci-data-science".

mrDzurb replied:
Good catch, thanks!
As for the ``%pip install llama-index-embeddings-oci-data-science``` i think the first one is the recommended way to install the package to the kernel without further reloading the kernel.

VipulMascarenhas reviewed (approved):
changes look good overall, minor comment on the notebook.

mrDzurb committed:
Fixes the notebook example.

mrDzurb committed:
Ruff fixes.

logan-markewich committed:
make openai content blocks optional (#17240)

mrDzurb committed:
Adjustments by black formatter.

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

zc277584121 committed:
refactor and optimize milvus code (#17229)

* refactor and optimize milvus code

Signed-off-by: ChengZi <chen.zhang@zilliz.com>

* Update pyproject.toml

---------

Signed-off-by: ChengZi <chen.zhang@zilliz.com>
Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

andrewisplinghoff committed:
Add async support to weaviate vector store integration (#17220)

* Add test_async_basic_flow()

* Make test_async_basic_flow() pass

* Make sync tests work again

* Make all tests pass

* Move client connect & close to fixture also in async tests

* Add adelete, adelete_nodes, aclear implementations

* Deduplicate code in query() / aquery()

* Get rid of clarified comment

This gets tested in `test_query_kwargs()`.

* Update WeaviateVectorStore documentation

* Remove from_params()

This method was not working as intended, it created a Weaviate v3 client instead of a v4 client as required by the rest of the code. It should be possible to just use the regular constructor instead of this method.

* Throw a custom exception when calling async methods without providing an async client

* Move AsyncClientNotProvidedError to a separate exceptions module

* Bump llama-index-vector-stores-weaviate version to 2.0.0

* Remove debug output

* DRY checks if _aclient is set, add SyncClientNotProvidedError

* Pass either sync or async client in `weaviate_client` parameter, fix connect()/close() when no weaviate client is provided

* Change new llama-index-vector-stores-weaviate version to 1.3.0 as the change is no longer breaking

* Downgrade to pytest 7 for compatibility with currently used pants configuration in CI

This commit can be reverted as soon as pytest >= 8 is used during the pants run.

* Reorganize test modules to prevent parallel execution during pants run

* Delete llama-index-integrations/vector_stores/llama-index-vector-stores-weaviate/poetry.lock

---------

Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

mdavidii committed:
[Issue #17219] MilvusVectorStore Parse "milvus_search_config" out of "vector_store_kwargs" (#17221)

* parse "milvus_search_config" out of "vector_store_kwargs" passed to MilvusVectorStore.query

* MilvusVectorStore Query parses "milvus_search_config" out of "vector_store_kwargs"

* use kwargs dict.get instead of named index;

* use kwargs.get for milvus_search_config throughout class;

* Update pyproject.toml

* pass **kwargs to _default_search

* linting

* actual linting

---------

Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

masci committed:
refact: Refactor `Document` to be natively multimodal (#17204)

* rename resource fields

* refactor Document

* fix typing, bring back text_template for backward compat

* fix bug in keyval docstore

* make TextNode forward-compatible

* redo deprecations

* fix model identifier

* update mocks

* update mocks

* fix fixture check

alexeyrodriguez committed:
fix(metrics): fixed NDCG calculation and updated previous tests (#17236)

masci committed:
fix: update unstructured dependency pin (#17246)

masci committed:
fix: accept already base64-encoded data in ImageBlock (#17244)

ashvin-a committed:
Add ILIKE feature for PGVectorStore and NileVectorStore. (#17211)

logan-markewich committed:
Fix content blocks again (#17247)

logan-markewich committed:
Add Scrapegraph tool integration (#17238)

masci committed:
feat: rewrite the example using AzureOpenAI (#17245)

Georgehe4 committed:
Handle empty retrieved Pinecone index values (#17242)

erika-shorten committed:
Update link for the Weaviate Python docs (#17248)

logan-markewich committed:
Gemini 2.0 support (#17249)

mrDzurb committed:
Fixed the notebook formatting.

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

seldo committed:
Adding llama 3.3-70b as function-calling-capable (#17253)

logan-markewich committed:
fix openai message dicts for tool calls (#17254)

xlinzh committed:
Fix: TGI context window (#17252)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

masci committed:
fix: make `ImageDocument` derive from `Document`, backward compatible (#17259)

* fix ImageDocument

* fix typing

* ImageDocument can be empty

* more coverage

* accept dict value in model constructor

* fix typing

* fix type check

* fix type check stronger

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

mrDzurb committed:
Modifies BUILD file. Adds ads mapping.

iamarunbrahma committed:
feat: Add NOT filter condition to MetadataFilter and QdrantVectorStore (#17270)

tomasonjo committed:
Add timeout config to neo4j graph (#17267)

fzowl committed:
Introducting VoyageAI's new multimodal embeddings model (#17261)

frangarcj committed:
User vs tool (#17273)

* fix: observation is tool response not user

* fix: observation is tool response not user

logan-markewich committed:
add content blocks to anthropic (#17274)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

logan-markewich committed:
Do not send blank content to anthropic (#17278)

logan-markewich committed:
fix links in RAG understanding (#17279)

logan-markewich committed:
add tool call id/name to mistral chat messages (#17280)

soulmachine committed:
Support Confluence cookies (#17276)

tomasonjo committed:
Add text and embedding type to neo4j enhanced schema (#17289)

MayRamati committed:
Update Tavily integration documentation (#17285)

fzowl committed:
VoyageAI multimodal embedding, correction (#17284)

jingyi-zhao-01 committed:
- add readme to google tools (#17275)

masci committed:
fix: add a timeout to langchain callback handler (#17296)

VinciGit00 committed:
ScrapegraphAI link was not correct and I fix it (#17292)

Update README.md

nerdai committed:
[bug fix] Ensure that `StopEvent` gets cleared from `Context._in_progress["_done"]` after a Workflow run (#17300)

* remove StopEvent from in progress

* add new and update stepwise test

mclean-connor committed:
Update anthropic type imports for v0.41.0 release (#17299)

Commit:
feat: Add tool support to AnthropicMultiModal (#17302)

schniber committed:
Amazon bedrock guardrails (#17281)

henrytwo committed:
Update Cerebras examples to use Llama 3.3 70b (#17301)

logan-markewich committed:
v0.12.6 (#17305)

logan-markewich committed:
o1 models (#17307)

* add latest o1 model support

* multimodal too

masci committed:
fix: make Document serialization backward compatible (#17312)

JeffreyEnglish committed:
CosmosDB insertion bugfix (#17290)

Adversarian committed:
Exposing additional keyword arguments for VoyageAI's embedding model (#17315)

Adversarian committed:
VoyageAI Reranker optional API Key (#17310)

ex0ns committed:
feat: add py.typed for OpenAI and Gemini (#17295)

mayconritzmann committed:
Update README.md (#17318)

logan-markewich committed:
restrict tests

amanyara committed:
Doc Error: modify part 2 of advanced prompts (#17255)

modify part2 of advanced prompts

Co-authored-by: zyh <2539110495@qq.com>

vntuananhbui committed:
Vntuananhbui multimodal rag guardrail (#17177)

logan-markewich committed:
wrong key name

jordanparker6 committed:
Feature/keywordsai llm (#16860)

LejianLeoHe7 committed:
Add OCI Generative AI tool calling support (#16888)

logan-markewich committed:
improve async search client handling (#17319)

logan-markewich committed:
v0.12.7 (#17320)

seldo committed:
NVIDIA Blueprint for content creation (#17327)

logan-markewich committed:
Tweak o1 function calling reqs (#17328)

seldo committed:
Fix NVIDIA Blueprint title (#17329)

Fix title

seldo committed:
Renamed, added badge (#17331)

mathematisse committed:
Fix missing aput in awrite_response_to_history (#17338)

dexters1 committed:
Cognee integration (#17314)

rachelbakkke committed:
Update API Reference and Component Guides For Sambanova (#17306)

Javtor committed:
Fix exclude text in document serialization (#17341)

* Fix text exclude in document serialization

* Fix test

* fix mypy

* fix imports

* pydantic bridge

* fix type

logan-markewich committed:
v0.12.8 (#17342)

shaidar committed:
Fix minor issues in rss (#17351)

fzowl committed:
example: VoyageAI's multimodal embeddings example (#17346)

dudanogueira committed:
implement client_kwargs["custom_batch"] for weaviate (#17347)

Aydin-ab committed:
bugfix when initializing with async aoss vector store (#17340)

xjtushilei committed:
TablestoreVectorStore check the Dimension of the embedding when writing it to store. (#17321)

wuxu92 committed:
`azureaisearch`: add semantic search mode support for async queries (#17335)

Riddhimaan-Senapati committed:
fixed the raptor example in llama-index-packs by updating the pip ins… (#17354)

masci committed:
chore: bump jinja version (#17361)

masci committed:
chore: bump jinja2 version in lockfiles (#17362)

fix: bump jinja2 version in lockfiles

schniber committed:
Fix Regression on Tools use for Bedrock Converse (#17364)

* fix #17317: corrected condition to include tools in boto3 client for Bedrock

* bump version of package

---------

Co-authored-by: Samuel CHNIBER <schniber@amazon.fr>

masci committed:
fix: prevent infinite recursion in get_article_urls (#17360)

* fix: prevent infinite recursion in get_article_urls

* bump version

mrkn committed:
Fix typo in property name (#17365)

* Fix typo in property name

Fixes the following error:

```
AttributeError: 'OpensearchVectorClient' object has no attribute 'os_version'
Python stacktrace:
 [1] _is_efficient_filtering_enabled
   @ llama_index.vector_stores.opensearch.base ~/.cache/pypoetry/virtualenvs/rag-company-rules-q2Dy5nHQ-py3.11/lib/python3.11/site-packages/llama_index/vector_stores/opensearch/base.py:637
 [2] __init__
   @ llama_index.vector_stores.opensearch.base ~/.cache/pypoetry/virtualenvs/rag-company-rules-q2Dy5nHQ-py3.11/lib/python3.11/site-packages/llama_index/vector_stores/opensearch/base.py:126
```

* Bump version

xjtushilei committed:
TablestoreVectorStore: support hybrid query, modify some documents. (#17366)

* TablestoreVectorStore: support hybrid query, modify some documents.

* update

okirmis committed:
Fix IndexError in LLM Reranking when handling malformed LLM responses (#17353)

* Fix bug #17352 by catching errors for malformed LLM responses

* Fix formatting

logan-markewich committed:
qdrant filter fix for `query_str is None` (#17377)

* remove unused query filter check

* remove unused query filter check

* fix bad merge

zc277584121 committed:
feat: milvus async (#17378)

yosuakoeswandy committed:
fix: Fix graphRAG notebook triplet extraction template (#17383)

Adversarian committed:
VoyageAIRerank constructor fix (#17343)

VinciGit00 committed:
feat: refactoring of example and readme (#17387)

nqtung committed:
fix storeindex cosmosnosql query issue - (BadRequest) One of the inpu… (#17385)

a-a-ronchen committed:
Pass aws_region_name to get_aws_service_client() in SageMakerLLM (#12000)

Commit:
update docstore.azurecosmosnosql pyproject.toml (#17391)

hashemian committed:
Fix: Properly add documents from Gitlab Repo (#17392)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

mrDzurb committed:
Removes ADS dependency.

logan-markewich committed:
clean up type hints in schema extractor (#17394)

* clean up type hints

* python does not have a good way to do this?

logan-markewich committed:
v0.12.9 (#17400)

dglogo committed:
Document Research Assistant for Blog Creation update (#17402)

* badge icon update

* launchable description

cjgatto committed:
Fix for llama_index.packs.raptor tree_traversal retrieval (#17406)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

Vlangf committed:
Add support for gpt-4o-2024-08-06 to multi modal llms (#17405)

Add support for gpt-4o-2024-08-06 in multi_modal_llms

raspawar committed:
NVIDIA Support for v2 Embedding & Reranking NIMs (#17410)

Commit:
feat: support typing.Annotated for adding parameter descriptions (#17411)

polarbear567 committed:
Fix bug 17398 (#17412)

logan-markewich committed:
vbump scrapegraph (#17418)

vbump

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

dglogo committed:
nvidia Document Research Assistant for Blog Creation ntbk arch diagram update (#17420)

arch diagram

dglogo committed:
updating notebook to automode off (#17421)

* arch diagram

* auto mode off

openvino-dev-samples committed:
update deps of OpenVINO packages (#17419)

* update hf hub deps

* update hf hub deps

nerdai committed:
Guess `image_mimetype` for `ImageNode` (#17422)

* add guess of mimetype from path to MediaResource

* test_build_image_node_image_resource and test_image_node_mimetype

* lint womp

* add missing assertion

logan-markewich committed:
remove asserts from schema llm extractor (#17425)

logan-markewich committed:
v0.12.10 (#17426)

ziodave committed:
remove unnecessary exclude line in pre-commit-config; update docs (#17431)

logan-markewich committed:
more exception handling in schema llm extractor (#17432)

logan-markewich committed:
unpin huggingface hub deps (#17433)

talibilat committed:
Corrected a mistake in the documentation (#17443)

Co-authored-by: Kashifa Sajjad <kashifasajjad@Kashifas-MBP.lan>

tomasonjo committed:
Fix neo4j schema bug (#17448)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

logan-markewich committed:
add request options to gemini (#17452)

* add request options to gemini

* update import

VinciGit00 committed:
feat: refactoring of examples for scrapegraphai (#17455)

inahjeon committed:
Update upstage solar model names and remove deprecated solar-docvision model (#17458)

raphael-foliveira committed:
adjust openai model temperature to accept values between 0 and 2 (#17453)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

inahjeon committed:
bugfix: Fix upstage embedding available model mapping (#17460)

shreyasbhs committed:
Added support to allow only public slack channels (#17399)

logan-markewich committed:
multimodal huggingface embeddings (#17463)

ziodave committed:
add example notebook; add quickstart link (#17466)

mjrowsky committed:
feat: async semantic splitter noder parser (#17449)

mathematisse committed:
Async memory management for OpenAIAgentWorker (#17375)

SuperSecureHuman committed:
Change top_n if number of input nodes are less (#17374)

masci committed:
feat: support adding images to ChatMessage through additional_kwargs (#17477)

shatejas committed:
Adds Opensearch in vector store integrations list (#17471)

Riddhimaan-Senapati committed:
Fixed issue #17397 by updating the markdown format in MarkdownReader (#17429)

logan-markewich committed:
note on contributing integrations (#17481)

minmie committed:
bugfix: Duplicated citation nodes (#17440)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

logan-markewich committed:
small nit on step decorator in notebook (#17486)

logan-markewich committed:
add test, fix missing method (#17485)

dglogo committed:
update language and embedding model (#17484)

dexters1 committed:
Cognee integration update (#17482)

rwaskiewicz committed:
docs: remove duplicate word (#17483)

masci committed:
docs: reword framework definition (#17498)

Necmttn committed:
Fix typo: Update knowledge_graph_rag_query_engine.ipynb (#17495)

ashvin-a committed:
Resolve issue with documentation in Nvidia multi modal notebook. (#17491)

ashvin-a committed:
Add new operator for Qdrant Vector store (#17490)

mi-iro committed:
Fix typos in docs transformations.md and PsychicDemo.ipynb (#17488)

xjtushilei committed:
Tablestore Support ChatStore/KvStore/DocStore/IndexStore. (#17473)

dnandha committed:
Add Multi Modal Bedrock Integration (#17451)

dexters1 committed:
Cognee integration update + notebook (#17510)

ashvin-a committed:
Add a required package in notebook for finetuning embeddings. (#17512)

logan-markewich committed:
handle pandas errors in md to df (#17511)

masci committed:
fix: bug in MediaResource validator (#17509)

* fix: bug in MediaResource validator

* fix logic

logan-markewich committed:
[WIP] Add support for OPEA LLMs in Llama-Index (#16666)

karsov committed:
Support MariaDB 11.7 in the MariaDB vector store integration (#17497)

logan-markewich committed:
v0.12.11 (#17524)

masci committed:
refact: make GeminiMultiModal a thin wrapper around Gemini (#17501)

fix unit test

fix request_options handling

save progress

address review comment

update notebook example

fix completion docs handling

KishalveP committed:
Deepseek-v3 is now supported by fireworks. (#17518)

* added deepseek-v3  model to fireworks

* bumped pyproject version

bestony committed:
feat: add stepfun integrations (#17514)

* feat: add stepfun integrations

* docs

* feat: add readme.md & fix error in BUILD

* feat: add model comment

* fix: fix line with newline at the end of file

* Apply suggestions from code review

---------

Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

NikhilShahi committed:
feat: Hyperbrowser Web Reader (#17489)

logan-markewich committed:
fix and update pants/cicd (#17538)

logan-markewich committed:
fix gigachat llm install (#17537)

ashishb committed:
feat: support direct PDF handling for Anthropic (#17506)

masci committed:
fix: align stepfun llm integration versioning (#17542)

logan-markewich committed:
MultiAgentWorkflow (#17237)

logan-markewich committed:
docs fixes (#17543)

docs

tituslhy committed:
Add longllmlingua2 integration (#17531)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

operagxsasha committed:
chore: docs readme (#17547)

david-oplatka committed:
API Migration (#17545)

chrisxiao committed:
fix typo (#17554)

moria97 committed:
Fix image-path validation in ImageNode (#17558)

* fix image path validation

* Update llama-index-core/llama_index/core/schema.py

Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

---------

Co-authored-by: Massimiliano Pippi <mpippi@gmail.com>

logan-markewich committed:
v0.12.12 (#17561)

tomasonjo committed:
Improves connections for neo4j objects and adds some tests (#17562)

cklapperich committed:
Contextual Retrieval (#17367)

vedantnaik19 committed:
Use correct colab url for Qdrant_metadata_filter (#17563)

logan-markewich committed:
get valid string when streaming (#17566)

masci committed:
fix: update vcrpy in vllm tests (#17572)

jayfish0 committed:
feat: Agentql Web Loader (#17575)

sourabhdesai committed:
Add framework integration for composite retrieval (#17536)

matea16 committed:
Vector index support for Memgraph's integration (#17570)

upchunk committed:
fix: :bug: [#17555] fix openai.BadRequestError: Invalid value for 'content': expected a string, got null. (#17556)

seldo committed:
Fix 2 links in agentworkflow notebook (#17591)

upchunk committed:
build: 🆙 replace `pinecone-client` with `pinecone` package (#17587)

logan-markewich committed:
support content blocks in chat templates (#17603)

leehuwuj committed:
fix reconstructing a tool in AgentWorkflow (#17596)

dependabot[bot] committed:
Bump JRubics/poetry-publish from 2.0 to 2.1 (#17589)

logan-markewich committed:
update function calling models in mistral (#17604)

juliette0704 committed:
add linkup tool (#17541)

jameslai-dev committed:
Made the message role of ReAct observation configurable (#17521)

tkreuder committed:
Add support for halfvec vector type (#17534)

minglu7 committed:
fix: correct the input params of "load_data" in NotionPageReader (#17529)

minglu7 committed:
refactor: Refactor markdown_to_tups method to better handle multi-lev… (#17508)

osamadel committed:
Advanced text to sql sample rows (#17479)

jpsoultanis committed:
Fixing header_path bug re: markdown level vs. stack depth (#17602)

logan-markewich committed:
v0.12.13 (#17605)

cgthayer committed:
Improving example code for Anthropic (#17607)

GinkREAL committed:
Fix prompt stacking in bedrock converse (#17613)

GinkREAL committed:
Implement async bedrock embeddings (#17610)

juliette0704 committed:
Add linkup tool (#17611)

Aktsvigun committed:
Nebius examples fixed (#17617)

logan-markewich committed:
DeepSeek official API LLM (#17625)

cmiller01 committed:
GoogleDriveReader support file extensions (#17620)

logan-markewich committed:
notebook nits (#17628)

logan-markewich committed:
Fix agentworkflow handoffs (#17631)

logan-markewich committed:
v0.12.14 (#17632)

LHFO94 committed:
.get_nodes now accepts include_values param to return embeddings (#17635)

advin4603 committed:
fix repeated sources when doing parallel tool calling (#17645)

rmacy committed:
small typo fix in the default plan refine prompt (#17644)

logan-markewich committed:
update llama-cpp integration + docs (#17647)

logan-markewich committed:
fix bedrock function calling (#17658)

KishalveP committed:
Deepseek-r1 is now supported by fireworks (#17657)

Riddhimaan-Senapati committed:
removed a dead link from fine-tuning.md in docs (#17652)

Javiery3889 committed:
[docs] Fix 404 link for Q&A Patterns (#17660)

LucasTonetto committed:
Fix: [Bug] NameError: name 'p' is not defined #15581 (#17577)

q-maze committed:
Add Snowflake Cortex Integration (#17585)

thommann committed:
Add endpoint parameter to TextEmbeddingsInference (#17598)

nerdai committed:
Remove API key and fix colab link in `deepseek.ipynb` (#17664)

nerdai committed:
[Fix] deps in `llama-index-networks/examples` (#17665)

antoineKorbit committed:
feat: allow to exclude empty file simple directory reader (#17656)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding

masci committed:
fix: make ctx._events_buffer json-serializable (#17676)

Javiery3889 committed:
[docs] Fix typo in BM25 Retriever Example (#17668)

alexander-azizi-martin committed:
Get description from pydantic field (#17679)

Riddhimaan-Senapati committed:
Fixed issue #17671 by adding the required decorators (#17678)

alexander-azizi-martin committed:
Add `error_on_tool_error` param to `FunctionCallingLLM.predict_and_call` (#17663)

edgBR committed:
Feat/fix Azure AI Search Hybrid Semantic Search Unusability due to hardcoded parameter (#17683)

Sylv-Lej committed:
Fix workflow module guide doc (#17686)

logan-markewich committed:
v0.12.15 (#17688)

logan-markewich committed:
o3 mini support (#17689)

logan-markewich committed:
Fix user_msg vs chat_history AgentWorkflow inputs (#17690)

logan-markewich committed:
tweak to deepseek function calling support (#17692)

jenperson committed:
Fixing broken link (#17691)

logan-markewich committed:
fix max_tokens, add reasoning_effort for openai reasoning models (#17694)

fix max_tokens, add reasoning_effort

logan-markewich committed:
small async voyageai fix (#17698)

mrDzurb committed:
Merge branch 'main' into ODSC-63451/oci_odsc_embedding
